hr_employee_relative/tests/test_hr_employee_relatives.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (C) 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from datetime import datetime
+
+from dateutil.relativedelta import relativedelta
+
+from odoo.tests import Form, common
+
+_ns = "hr_employee_relative"
+
+
+class TestHrEmployeeRelatives(common.TransactionCase):
+    def setUp(self):
+        super().setUp()
+        self.Employee = self.env["hr.employee"]
+        self.EmployeeRelative = self.env["hr.employee.relative"]
+
+    def test_age_calculation(self):
+        employee = self.Employee.create(
+            {
+                "name": "Employee",
+                "relative_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "relation_id": self.env.ref(_ns + ".relation_sibling").id,
+                            "partner_id": self.env.ref("base.res_partner_1").id,
+                            "name": "Relative",
+                            "date_of_birth": datetime.now() + relativedelta(years=-42),
+                        },
+                    )
+                ],
+            }
+        )
+        relative = self.EmployeeRelative.browse(employee.relative_ids[0].id)
+        self.assertEqual(int(relative.age), 42)
+        # onchange partner
+        ctx = {
+            "active_ids": [relative.id],
+            "active_id": relative.id,
+            "active_model": "hr.employee.relative",
+        }
```

Comment:
Is this really needed?
When using Form in line 46 it will try to open any editable view of hr.employee.relative. And we have defined the tree view as editable, and it has partner_id and relation_id
No need at all then to do this. By default it will load that view and will be able to apply the onchanges there. 

with Form(relative) as f: should be enough